modules/economy-system/economy-system.js

```diff
@@ -124,67 +125,274 @@ async function editBank(client, id, action, value) {
-            resolve(`Deleted the item ${item} successfully`);
+            if (!client) return resolve('Please pass a client');
+            const model = await client.models['economy-system']['Shop'].findAll({
+                where: {
+                    [Op.or]: [
+                        {name: pName},
+                        {id: pId}
+                    ]
+                }
+            });
+            if (model.length > 1) {
+                resolve('More than one item was found');
+            } else if (model.length < 1) {
+                resolve('No item was found');
+            } else {
+                await model[0].destroy();
+                client.logger.info(`[economy-system] ` + localize('economy-system', 'delete-item', {
+                    u: 'API/ CLI',
+                    i: pName
+                }));
+                if (client.logChannel) client.logChannel.send(`[economy-system] ` + localize('economy-system', 'delete-item', {
+                    u: 'API/ CLI',
+                    i: pName
+                }));
+                resolve(`Deleted the item ${pName}/ ${pId} successfully`);
+            }
         }
     });
 }
 
 /**
  * Create the shop message
  * @param {Client} client Client
+ * @param {object} guild Object of the guild
+ * @param {boolean} ephemeral Should the message be ephemeral?
  * @returns {string}
  */
-async function createShopMsg(client) {
+async function createShopMsg(client, guild, ephemeral) {
     const items = await client.models['economy-system']['Shop'].findAll();
     let string = '';
+    const options = [];
     for (let i = 0; i < items.length; i++) {
-        string = `${string}**${items[i].dataValues.name}**: ${items[i].dataValues.price} ${client.configurations['economy-system']['config']['currencySymbol']}\n`;
+        const roles = await guild.roles.fetch(items[i].dataValues.role);
+        string = `${string}${inputReplacer({'%id%': items[i].dataValues.id, '%itemName%': items[i].dataValues.name, '%price%': `${items[i].dataValues.price} ${client.configurations['economy-system']['config']['currencySymbol']}`, '%sellcount%': roles.members.size}, client.configurations['economy-system']['strings']['itemString'])}`;
+        options.push({
+            label: items[i].dataValues.name,
+            description: localize('economy-system', 'select-menu-price', {
+                p: `${items[i].dataValues.price} ${client.configurations['economy-system']['config']['currencySymbol']}`
+            }),
+            value: items[i].dataValues.id
+        });
     }
-    return embedType(client.configurations['economy-system']['strings']['shopMsg'], {'%shopItems%': string}, { ephemeral: true });
+    const components = [{
+        type: 'ACTION_ROW',
+        components: [{
+            type: 3,
+            placeholder: localize('economy-system', 'nothing-selected'),
+            'min_values': 1,
+            'max_values': 1,
+            options: options,
```

Comment:
This is an empty array if no items are in the shop, causing a crash on message send